src/libraries/System.Private.CoreLib/src/System/Type.cs

```diff
@@ -438,8 +438,52 @@ public static Type[] GetTypeArray(object[] args)
             return cls;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static TypeCode GetTypeCode(Type? type)
         {
+            if (RuntimeHelpers.IsKnownConstant(type))
+            {
+                // keep in sync with RuntimeType.cs
```

Comment:
Would it be better to actually share the code instead?

Change this to:
```
if (RuntimeHelpers.IsKnownConstant(type) && type is RuntimeType)
{
    return ((RuntimeType)type).GetTypeCodeInternal();
}
```
And change RuntimeType.GetTypeCodeImpl to: 
```
protected override TypeCode GetTypeCodeImpl()
{
    TypeCode typeCode = Cache.TypeCode;

    if (typeCode != TypeCode.Empty)
        return typeCode;

    return Cache.TypeCode = GetTypeCode();
}
```